lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -472,7 +472,12 @@ def _on_opensearch_data_storage_detaching(self, _: StorageDetachingEvent):  # no
                 "Removing units during an upgrade is not supported. The charm may be in a broken, unrecoverable state"
             )
         # acquire lock to ensure only 1 unit removed at a time
-        if not self.node_lock.acquired:
+        # Closes canonical/opensearch-operator#378
+        if (
+            self.app.planned_units() > 1
+            and (self.opensearch.is_node_up() or self.alt_hosts)
+            and not self.node_lock.acquired
+        ):
```

Comment:
So that means we are planning to no longer use the locking mechanism if `planned_units()` is 0 or 1?

Reply:
Yeah, but we may have to add: `online_nodes > 1 and self.app.planned_units() > 1`. This way we have a single node in a cluster not expected to grow...

Reply:
@reneradoi check my comment answer [here](https://github.com/canonical/opensearch-operator/pull/361/files/e3588a23397c17f357a1d5ec3254958bca366fa1#r1711549293)